### PR TITLE
Flake fixes

### PR DESCRIPTION
**Pull request by Gundersanne**
number tests succeeded: 6
failures: 1, on clicking run which would indicate a podman error

### CODE REVIEW

File: lib/cockpit-components-file-autocomplete.jsx [hunk 1]
```diff
@@ -53,12 +53,13 @@ export class FileAutoComplete extends React.Component {
...
-            if (!this.updateIfDirectoryChanged(value)) {
+            this.updateIfDirectoryChanged(value, () => {
+                this.pendingCallback = false;
                 let stateUpdate = this.filterFiles(value);
-                this.setState(stateUpdate);
-                this.onChangeCallback(value, { error: stateUpdate.error });
-            }
+                this.setState(stateUpdate, () => {
+                    this.onChangeCallback(value, { error: stateUpdate.error });
+                });
```
Comment (martinpitt) [inline lib/cockpit-components-file-autocomplete.jsx]: No need for the { }.

File: lib/cockpit-components-file-autocomplete.jsx [hunk 2]
```diff
@@ -153,16 +154,18 @@ export class FileAutoComplete extends React.Component {
...
         const changed = directory !== this.state.directory;
         if (changed && this.state.directoryFiles !== null) {
-            this.setState({
-                displayFiles: [],
-                directoryFiles: null,
-                directory: directory,
-                open: false,
-            });
+            this.setState(() => {
+                return {
```
Comment (martinpitt) [inline lib/cockpit-components-file-autocomplete.jsx]: No need for the `{ return; }`

**Comment by martinpitt** [comment]
Please also add a Closes #.

This looks like it would equally apply to cockpit itself, can you please send a PR? Thanks!

File: lib/cockpit-components-file-autocomplete.jsx (hunk 2 above)
Comment (Gundersanne) [reply lib/cockpit-components-file-autocomplete.jsx]: In this case it's needed i think, as that's the functional version.

Comment (martinpitt) [reply lib/cockpit-components-file-autocomplete.jsx]: Right, but `() => { foo: null }` is still a function. :-)

Comment (martinpitt) [reply lib/cockpit-components-file-autocomplete.jsx]: In general, `() => expression` is fully equivalent to `() => { return expression }`

**Pull request merged**